vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/SlideManagerTest.java

```diff
@@ -0,0 +1,85 @@
+package edu.asu.diging.vspace.core.services.impl;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+import edu.asu.diging.vspace.core.data.SequenceRepository;
+import edu.asu.diging.vspace.core.data.SlideRepository;
+import edu.asu.diging.vspace.core.exception.SlideDoesNotExistException;
+import edu.asu.diging.vspace.core.model.ISlide;
+import edu.asu.diging.vspace.core.model.impl.Sequence;
+import edu.asu.diging.vspace.core.model.impl.Slide;
+
+public class SlideManagerTest {
+
+    @Mock
+    private SequenceRepository sequenceRepo;
+    
+    @Mock
+    private SlideRepository slideRepo;
+    
+    @InjectMocks
+    private SlideManager slideManagerToTest = new SlideManager();
+    
+    @InjectMocks
+    private SlideManager slideManagerToTestSpy =  Mockito.spy(slideManagerToTest);
+    
+    @Before
+    public void init() {
+        MockitoAnnotations.initMocks(this);
+    }
+    
+    @Test
+    public void test_slideSequence() {
+        String slideId = "SLI000000002";
+        String moduleId = "MOD000000002";
+        List<ISlide> slidesList = Arrays.asList(new Slide());
+        slidesList.get(0).setId("SLI000000002");
+        
+        Sequence sequenceObj = new Sequence();
+        sequenceObj.setId("SEQ000000004");
+        sequenceObj.setSlides(slidesList);
+        List<Sequence> sequencesList = new ArrayList<>();
+        sequencesList.add(sequenceObj);
+        Mockito.when(sequenceRepo.findSequencesForModule(moduleId)).thenReturn(sequencesList);
+        
+        List<Sequence> actualSequenceSlideList = slideManagerToTest.getSlideSequences(slideId, moduleId);
+        
+        Assert.assertEquals(actualSequenceSlideList.get(0).getSlides().get(0).getId(), sequencesList.get(0).getSlides().get(0).getId());
+        Assert.assertEquals(actualSequenceSlideList.size(), sequencesList.size()); 
+    }
+    
+    @Test
+    public void test_deleteSlideById() throws SlideDoesNotExistException {
+        String slideId = "SLI0000000040";
+        String moduleId = "MOD000000010";
+
+        List<ISlide> slidesList = Arrays.asList(new Slide());
+        slidesList.get(0).setId("SLI0000000040");
+        
+        Sequence sequenceObj = new Sequence();
+        sequenceObj.setId("SEQ000000004");
+        sequenceObj.setSlides(slidesList);
+        List<Sequence> sequencesList = new ArrayList<>();
+        sequencesList.add(sequenceObj);
+        Mockito.when(sequenceRepo.findSequencesForModule(moduleId)).thenReturn(sequencesList);
+        
+        ISlide slide = new Slide();
+        slide.setId(slideId);
+        Mockito.doReturn(slide).when(slideManagerToTestSpy).getSlide(Mockito.any());
+        
+        slideManagerToTestSpy.deleteSlideById(slideId, moduleId);  
```

Comment:
you don't need a spy object here, just mock the getById call on the repository. This way even if the implementation changes and the delete method calls the repository directly, the test won't fail.

Reply:
Without using spy, I tried to call using the repo directly, but it threw an exception as null data was getting returned.
To mock methods in the same test class, answers in the net suggested using spy object.

Reply:
I'm not 100% what you mean but if you mock `getById` (or whicheever method is called by `getSlide`) to return something, it won't throw a null pointer.